### PR TITLE
W2D: Update empty states

### PR DESCRIPTION
**Pull request by surags**
https://rally1.rallydev.com/#/357251704080ud/custom/367300408400?detail=%2Fuserstory%2F483243445644&fdp=true

image
image
image

### CODE REVIEW

File: components/d2l-work-to-do/d2l-work-to-do.js [hunk 1]
```diff
@@ -91,16 +92,13 @@ class WorkToDoWidget extends EntityMixinLit(LocalizeWorkToDoMixin(LitElement)) {
...
-					width: 100px;
+					max-width: 18rem;
+					width: 100%;
 				}
 				.d2l-work-to-do-fullscreen-container {
 					background-image: linear-gradient(to bottom, #f9fbff, #ffffff);
 					padding: 0 2rem;
 				}
-				d2l-button {
-					padding: 1.8rem 0;
```
Comment (mbardakovD2L) [inline components/d2l-work-to-do/d2l-work-to-do.js:102]: is this just for the "load more" button? does this interfere with the non-empty state at all?

File: components/d2l-work-to-do/d2l-work-to-do.js [hunk 2]
```diff
@@ -220,34 +218,56 @@ class WorkToDoWidget extends EntityMixinLit(LocalizeWorkToDoMixin(LitElement)) {
...
+					</div>
+					<div class="d2l-body-standard d2l-empty-body-text-container">
+						${this.localize('comeBackNoFutureActivities')}
+					</div>
+				`;
 			}
 
-			return discoverActive
-				? html`${this.localize('noActivitiesDiscoverActive')}`
-				: html`${this.localize('noActivitiesDiscoverInactive')}`;
```
Comment (mbardakovD2L) [inline components/d2l-work-to-do/d2l-work-to-do.js:230]: do we no longer have a concept of "you have no activities, use discover to find more"?

File: components/d2l-work-to-do/d2l-work-to-do.js [hunk 3]
```diff
@@ -256,19 +276,14 @@ class WorkToDoWidget extends EntityMixinLit(LocalizeWorkToDoMixin(LitElement)) {
 			}
 
 			return html`
+				${this.fullscreen ? immersiveNav() : ''}
```
Comment (mbardakovD2L) [inline components/d2l-work-to-do/d2l-work-to-do.js:279]: I _think_ it'd be easier/cleaner to handle the `this.fullscreen?` logic inside `immersiveNav()` itself, i.e. if we're not in fullscreen view then `immersiveNav()` would return the `nothing` directive

File: components/d2l-work-to-do/lang/en.js [hunk 4]
```diff
@@ -12,12 +14,16 @@ export const val = {
 	loadMore : "Load More", // Button text displayed in 'Fullscreen View' that allows the user to access the next page of activities which will append to the bottom of the list currently shown
 	loadMoreDescription : "Display more assigned activities", // Additional description text to accompany the load more button for additional clarity for the user
 	myWorkToDo: "My Work To Do", // Widget title
+	noActivities : "You have no incomplete activities with due or end dates available.", // 'Empty state' - When widget has no activities in full page view
 	noActivitiesDiscoverActive : "You have no activities due or ending in the next two weeks. Check out Discover to find more learning.", // 'Empty View' - When widget has no activities to display within the next two weeks, but discover is active and can be navigated to by interacting with the provided element
 	noActivitiesDiscoverInactive : "You have no activities with due or end dates available.", // 'Empty View' - When widget has no activities to display within the next two weeks & there are no more activities in the future & discover is not active for this user
```
Comment (mbardakovD2L) [inline components/d2l-work-to-do/lang/en.js:19]: if we _are_ getting rid of the "go to discover" linking we could also clean up these two langterms

File: components/d2l-work-to-do/d2l-work-to-do.js (hunk 1 above)
Comment (surags) [reply components/d2l-work-to-do/d2l-work-to-do.js:102]: This change was intended for the view all work button. The only other place d2l-button is used is the load more button and I have updated the styling of the load more class to reflect the change

File: components/d2l-work-to-do/d2l-work-to-do.js (hunk 2 above)
Comment (surags) [reply components/d2l-work-to-do/d2l-work-to-do.js:230]: The code seems to make references to discover but none if it is ever used. This was my attempt to clean it up a little. We can add it back when we work on adding discover. I don't feel strongly about it so I can revert if needed

File: components/d2l-work-to-do/d2l-work-to-do.js (hunk 3 above)
Comment (surags) [reply components/d2l-work-to-do/d2l-work-to-do.js]: Updated

File: components/d2l-work-to-do/d2l-work-to-do.js (hunk 2 above)
Comment (mbardakovD2L) [reply components/d2l-work-to-do/d2l-work-to-do.js:230]: right - that makes sense. I'm also good either way but we should ask product/design if that's still planned

File: components/d2l-work-to-do/d2l-work-to-do.js [hunk 5]
```diff
@@ -256,19 +279,14 @@ class WorkToDoWidget extends EntityMixinLit(LocalizeWorkToDoMixin(LitElement)) {
 			}
 
 			return html`
+				${immersiveNav()}
```
Comment (JasperAtD2L) [inline components/d2l-work-to-do/d2l-work-to-do.js:282]: It's weird that we have empty/fullscreen/and widget (activity) states. It should really be empty/activities with a modifier of fullscreen. That being said we can't make that change.

Where I am going is; the immersiveNav is a full screen only thing, so the choice to display the immersive nav should be made here and not in the immersiveNav function.

File: components/d2l-work-to-do/d2l-work-to-do.js (hunk 3 above)
Comment (mbardakovD2L) [reply components/d2l-work-to-do/d2l-work-to-do.js:282]: update: nevermind, see Jasper's comments below

**Review by JasperAtD2L — APPROVED** [review]

**Review by mbardakovD2L — APPROVED** [review]

**Pull request merged**